src/mlpack/methods/ann/layer/linear_impl.hpp

```diff
@@ -41,6 +41,62 @@ Linear<InputDataType, OutputDataType, RegularizerType>::Linear(
   weights.set_size(outSize * inSize + outSize, 1);
 }
 
+template<typename InputDataType, typename OutputDataType,
+    typename RegularizerType>
+Linear<InputDataType, OutputDataType, RegularizerType>::Linear(
+	const Linear& layer) :
+    inSize(layer.inSize),
+    outSize(layer.outSize),
+    weights(layer.weights),
+    regularizer(layer.regularizer)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType,
+    typename RegularizerType>
+Linear<InputDataType, OutputDataType, RegularizerType>::Linear(
+  Linear&& layer) :
+    inSize(layer.inSize),
+    outSize(layer.outSize),
+    weights(std::move(layer.weights)),
+    regularizer(std::move(layer.regularizer))
+{
+  // Nothing to do here.
```

Comment:
I think we should set `layer.inSize` and `layer.outSize` to 0 here, since we are taking ownership of everything in the other `Linear` layer, and after this is called, it should be an "empty" layer.